src/NServiceBus.Transport.Msmq/MessagePump.cs

```diff
@@ -30,15 +30,20 @@ public void Dispose()
         {
             peekCircuitBreaker?.Dispose();
             receiveCircuitBreaker?.Dispose();
-            cancellationTokenSource?.Dispose();
+            messagePumpCancellationTokenSource?.Dispose();
```

Comment:
is it intentional to dispose this twice? It seems to be already disposed in `StopReceive`.

Reply:
totally missed that we implemented IDisposable, will fix